packages/ui/components/form/select/select.test.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import { vi } from "vitest";
+
+import { Select, SelectField, SelectWithValidation } from "./Select";
+
+const options = [
+  { label: "Option 1", value: "option1" },
+  { label: "Option 2", value: "option2" },
+  { label: "Option 3", value: "option3" },
+];
+
+const onChangeMock = vi.fn();
+
+let props: any = {
+  name: "test",
+  options: options,
+  defaultValue: { label: "Option 3", value: "option3" },
+  onChange: onChangeMock,
+};
+
+const renderSelectWithForm = () => {
+  render(
+    <form aria-label="test form">
+      <label htmlFor="test">Test</label>
+      <Select {...props} inputId="test" />
+      <p>Click Outside</p>
+    </form>
+  );
+};
+
+const selectOption = async (optionText: any) => {
+  const element = screen.getByLabelText("Test");
+  element.focus();
+  fireEvent.keyDown(element, { key: "ArrowDown", code: "ArrowDown" });
+  screen.getByText(optionText);
+  const option = screen.getByText(optionText);
+  fireEvent.click(option);
+};
+
+describe("Tests for Select File", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe("Tests for Select component", () => {
+    beforeEach(() => {
+      renderSelectWithForm();
+    });
+
+    test("Should render with the correct default value", () => {
+      expect(screen.getByText("Option 3")).toBeInTheDocument();
+      expect(screen.getByRole("form")).toHaveFormValues({ test: "option3" });
+    });
+
+    test("Should select and have the correct value after selection of an option", async () => {
+      await waitFor(async () => {
+        await selectOption("Option 2");
+      });
+
+      expect(onChangeMock).toHaveBeenCalled();
+      expect(screen.getByRole("form")).toHaveFormValues({ test: "option2" });
+    });
+
+    test("Should keep the default value after selections are shown and the user clicks outside the selection element", async () => {
+      await waitFor(async () => {
+        const element = screen.getByLabelText("Test");
+        element.focus();
+        fireEvent.keyDown(element, { key: "ArrowDown", code: "ArrowDown" });
+
+        screen.getByText("Option 2");
+        const outsideButton = screen.getByText("Click Outside");
+        fireEvent.click(outsideButton);
+      });
+
+      expect(screen.getByRole("form")).toHaveFormValues({ test: "option3" });
+    });
+
+    test("Should keep the selected value after the user has selected an option and clicked out of the selection element", async () => {
+      await waitFor(async () => {
+        await selectOption("Option 2");
+        const outsideButton = screen.getByText("Click Outside");
+        fireEvent.click(outsideButton);
+      });
+
+      expect(screen.getByRole("form")).toHaveFormValues({ test: "option2" });
+    });
+  });
```

Comment:
We don't seem to be testing the code that we have written in `Select` component. 
In the component, we are just configuring the props for react-select.
<img width="1166" alt="Screenshot 2023-07-31 at 12 45 44 PM" src="https://github.com/calcom/cal.com/assets/1780212/6e2d0ec9-6d5a-437d-a95d-f9393c6dfdd5">

- We need a test for verifying that any classes passed to Select are passed to ReactSelect
- We might need to test certain classes as well. It isn't recommended to test implementation details so may be we keep a color map for tests and verify the background color of the DOM element is CAL_BG_SUBTLE when isDisabled prop is given.
- Similarly we should probably test the result of passing all these props - isFocussed, isDisabled, isSelected. It might be overkill as well